### PR TITLE
fix build errors and add postReceipt PostingTransactionsTests

### PR DESCRIPTION
**Pull request by beylmk**
* fix a couple build errors in unit tests
* updates `PostingTransactionTests` to use new stubStoreProduct method
* adds a couple missing tests to `PostingTransactionsTests`, including one for pricing phases being sent
* uses `stubStoreProduct` in a few leftover PurchasesTests

### CODE REVIEW

File: purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt [hunk 1]
```diff
@@ -171,30 +173,19 @@ class PostingTransactionsTests {
             onError = { _, _ -> }
         )
         assertThat(postedProductInfoSlot.isCaptured).isTrue
-        assertThat(postedProductInfoSlot.captured.duration).isEqualTo(expectedSubscriptionPeriod)
-        assertThat(postedProductInfoSlot.captured.introDuration).isEqualTo(expectedIntroPricePeriod)
-        assertThat(postedProductInfoSlot.captured.trialDuration).isEqualTo(expectedFreeTrialPeriod)
-        verify(exactly = 1) {
```
Comment (beylmk) [inline purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt:177]: this `verify` was called in a lot of unit tests where it didn't feel necessary

Comment (tonidero) [reply purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt:177]: Hmm I guess it makes sure we are only posting to the backend once... I think it's ok to remove it, but we probably want to have a test to verify that we don't post multiple times.

File: purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt [hunk 2]
```diff
@@ -171,30 +173,19 @@ class PostingTransactionsTests {
...
-                onSuccess = any(),
-                onError = any()
-            )
-        }
+        assertThat(postedProductInfoSlot.captured.duration).isEqualTo(mockStoreProduct.subscriptionPeriod)
     }
 
     @Test
-    fun `inapps send null durations when posting to backend`() {
+    fun `purchaseOptionId is sent when posting to backend`() {
```
Comment (tonidero) [inline purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt:180]: Should we verify the pricing phases are correct in another test as well? We can add that in another PR but I would add a TODO in that case so we don't forget.

Comment (beylmk) [reply purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt:180]: Yes definitely!  I have another branch started w pricing phase validation, in the BackendTests

File: purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt (hunk 1 above)
Comment (beylmk) [reply purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt:177]: good point, i just think one test should test one thing...but let me add one for that case!

Comment (beylmk) [reply purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt:177]: okay, actually, testing it's call just once happens for any of the tests where we validate certain parameters going through to backend, so i think that's covered.

**Pull request merged**